### ISSUE TITLE
PyTorch Implementation - Reproducibility

### ISSUE THREAD

**Issue by FabioArnez** [microsoft/AirSim-Drone-Racing-VAE-Imitation#5]
Hi,

congrats on this amazing work. I was trying to reproduce your results using PyTorch. However, I was not able to get the same results for the CM-VAE&control policy, using the parameters in the code and the training details reported in the paper. Below, I summarize the problems I encountered:

1. For CMVAE Unconstrained:
    - Poor reconstruction quality when using the loss weights (w_img, w_gate)  and beta as in **train_cmvae.py**. With this CMVAE performance, I was not able to train a good downstream control policy.
    - I improved the reconstruction quality by increasing the loss weights (w_img, w_gate, kept beta=8) and changing dataset splits from 80%-20% to 90%-10% for training and testing respectively. With these changes, I successfully trained the downstream policy. However, the navigation policy performance is far from the results reported in the paper ( Fig. 7), especially when random noise for gate offset is higher than 1.

2. For CMVAE Constrained:
    - Same situation as in the unconstrained version. Improved the results increasing the loss weights as before, but I'm not getting a disentangled representation.
    - The navigation policy performance is similar to the unconstrained version (not good when random noise for gate offset is higher than 1).

I was wondering if you used fixed loss weights&beta during training? or if dynamic values were used during training (e.g. beta annealing, ...)?. Could you provide more training details about **CM-VAE constrained**?

In addition, is it possible to have the random seeds for the generated tracks/gates used in the navigation-policy performance assessment (Fig. 7)? 

Thank you so much for your help! and again congrats, impressive work!

**Comment by rogeriobonatti** [comment]
Hi Fabio, thanks for the message! In theory I see no issues with the results being reproduced in another deep learning framework. A few comments about what you described:

- I did play around a lot with relative weights between losses when I was developing this paper to ensure that the image reconstruction was visually good, and at the same time that the gate pose estimation was also good. 

- On the gate pose estimate, sometimes the "yaw" of the gate was not learned (in about 1/10 of times that I trained the network), probably due to random initialization factors -- just re-training with a new random initialization worked. 

- I did not use beta annealing in the final results, but I did play around with it a few times. It's worth a try, and it might improve final results.

- The seeds that I used are likely the same ones that are in the current files in this repo

- As for your performance issues, I encourage you double-check all details of your network and data pipelines. Visualizing the representations and gate estimates helps a lot to understand what can be happening. 

If you have any more questions please post them here! And also send a link to your Pytorch implementation once you're done :)
